instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt

```diff
@@ -1,321 +1,331 @@
+        sdkTest {
+            setupMockRequestHandlers()
+            withMapboxNavigation(
+                historyRecorderRule = mapboxHistoryTestRule
+            ) { mapboxNavigation ->
+                val testRoutes = mapboxNavigation.requestNavigationRoutes(startCoordinates)
+                val originOfTestRoute = testRoutes.first().routeOptions.coordinatesList().first()
+                stayOnPosition(
+                    latitude = originOfTestRoute.latitude(),
+                    longitude = originOfTestRoute.longitude(),
+                    bearing = 30.0f
+                ) {
+                    mapboxNavigation.startTripSession()
+
+                    val alternativesCallbackResultBeforeSetRoute =
+                        firstAlternativesUpdateDeferred(mapboxNavigation)
+                    mapboxNavigation.setNavigationRoutesAsync(testRoutes)
+                    mapboxNavigation.routeProgressUpdates().first()
+                    val alternativesCallbackResultAfterSetRoute =
+                        firstAlternativesUpdateDeferred(mapboxNavigation)
 
-        runBlocking(Dispatchers.Main) {
-            val historyPath = mapboxNavigation.historyRecorder.stopRecording()
-            logE("history path=$historyPath", LOG_CATEGORY)
+                    assertTrue(
+                        "the test expects that alternative routes are present",
+                        mapboxNavigation.getNavigationRoutes().size > 1
+                    )
+                    assertTrue(alternativesCallbackResultBeforeSetRoute.isActive)
+                    alternativesCallbackResultBeforeSetRoute.cancel()
+                    assertTrue(alternativesCallbackResultAfterSetRoute.isActive)
+                    alternativesCallbackResultAfterSetRoute.cancel()
+                }
+            }
         }
 
-        // Verify alternative routes events were triggered.
-        assertEquals(2, routes.size)
-        assertTrue(mapboxNavigation.getRoutes().isNotEmpty())
-        firstAlternative.verifyOnRouteAlternativesAndProgressReceived()
-
-        assertNotEquals(
-            "Verify setRoutes and alternatives are not the same routes",
-            routes.drop(1).sortedBy { it.hashCode() }, // drop primary route
-            firstAlternative.alternatives!!.sortedBy { it.hashCode() }
-        )
-    }
-
     /**
-     * The test verifies that if set routes to Navigation that are come from alternatives + n
-     * (where n >= 1) external routes, alternatives routes observer is not triggered with these routes.
-     * That was happening in legacy versions of NN.
+     * The fact that the SDK triggers a callback on alternatives subscription I consider as a bug:
+     * 1. We recommend users to set updated alternatives back, and it doesn't make sense to trigger
+     * the callback with the same alternative as they are now.
+     * 2. The SDK doesn't trigger callback if routes were set already, so in general it feels
+     * inconsistent
```

Comment:
@dzinad , @DzmitryFomchyn , can you please share your opinion about this SDK behaviour. Does it seem like a bug to you? Do you think I should open a ticket for this or it's better to solve it in this PR?

Reply:
Agreed. I actually discussed the similar thing with NN and it seems like they shouldn't return alternatives, I'll send you the link in DM, could you verify if that's the same thing?